flujo/application/core/executor_core.py

```
@@ -505,6 +509,52 @@ def _update_context_state(self, context: Optional[TContext_w_Scratch], state: st
+
+        if isinstance(context, BaseModel):
+            for field_name, field_value in context:
+                if isinstance(field_value, ContextReference):
+                    # Only save if value is present (hydrated/modified)
+                    # Accessing _value directly to avoid ValueError from get()
+                    if field_value._value is not None:
+                        provider = self._state_providers.get(field_value.provider_id)
+                        if provider:
+                            try:
+                                await provider.save(field_value.key, field_value._value)
+                            except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)